### ISSUE TITLE
Allow to set database name when using JDBC URL

### ISSUE THREAD

**Issue by manikmagar** [testcontainers/testcontainers-java#566]
As mentioned [here](https://www.testcontainers.org/usage/database_containers.html#jdbc-url) in documentation, when using JDBC URL to initialize database, name is ignored and defaulted to test. Some times it is required to specify the database name.

Please make JDBC drivers to honor the database name set in the URL.

For example, `jdbc:tc:mysql://somename:someport/library?TC_INITSCRIPT=library_db.sql` should initialize a database named library and not test.

**Comment by bopfermann** [comment]
+1

The only way I found around this issue was to create custom extensions of JdbcDatabaseContainer with the appropriate hard coded value.

In my case this was more or less a copy-paste of the source for MySQLContainer

Requires a custom 
``` /META-INF/services/org.testcontainers.containers.JdbcDatabaseContainerProvider ```
with the fully qualified classname of your custom implementation.

**Comment by kiview** [comment]
Hey @bopfermann,
would you like to provide a PR implementing this feature request? :slightly_smiling_face:

**Comment by manikmagar** [comment]
I think JDBCDatabaseContainerProvider should be modified [here#L10](https://github.com/testcontainers/testcontainers-java/blob/master/modules/jdbc/src/main/java/org/testcontainers/containers/JdbcDatabaseContainerProvider.java#L10)  -

`public abstract JdbcDatabaseContainer newInstance(String tag, String jdbcUrl);`

So, when a new container instance is acquired [here](https://github.com/testcontainers/testcontainers-java/blob/master/modules/jdbc/src/main/java/org/testcontainers/jdbc/ContainerDatabaseDriver.java#L122), a reference of JDBC URL can be passed to the provider.

Then container providers such as MySQLContainerProvider [here](https://github.com/testcontainers/testcontainers-java/blob/master/modules/mysql/src/main/java/org/testcontainers/containers/MySQLContainerProvider.java#L14) can utilize details from the URL and instantiate database with given name/details.

That is what is in mind for solution, Does that make sense?

**Comment by kiview** [comment]
Sounds like the right approach, but would be an API breaking change. Would you like to submit a PR and we look into the consequences?

**Comment by manikmagar** [comment]
Yes, it breaks the API for container provider. Sure, let me give it a try and then we can verify the behavior. Thanks.

**Comment by manikmagar** [comment]
Thinking out loud, how about providing an overloaded signature for newInstance method? With that, those providers who don’t need to implement this can still retain current behavior. WDYT?

**Comment by kiview** [comment]
And this method could have a default implementation returning `newInstance(tag)`?

**Comment by manikmagar** [comment]
Added a PR #594 for this implementation.
